### PR TITLE
Popup window and dynamic cards feature

### PR DESCRIPTION
**Pull request by Rebelzob**
# Portfolio

- Complete rework of card sections in HTML file, added Javascript to populate them in the HTML dynamically.
- Added Dialog tag to HTML and worked on it for the modals.
- Added modal styling for mobile and desktop versions
- Added Javascript logic to populate modals with the cards object.

### CODE REVIEW

**Review by GzimAsani — CHANGES_REQUESTED** [review]
Hi @Rebelzob , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that does not have commits from your pair programming partner

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Comment by Rebelzob** [comment]
@GzimAsani forgot to add to the description that **I don't have** a coding partner this week, already reported the issue to student success and they told me to work alone.

**Comment by GzimAsani** [comment]
> @GzimAsani forgot to add to the description that **I don't have** a coding partner this week, already reported the issue to student success and they told me to work alone.

Please mention and add a screenshot of it

**Comment by Rebelzob** [comment]
@GzimAsani This is the email I got from the student success area.
image

**Comment by GzimAsani** [comment]
> @GzimAsani This is the email I got from the student success area.
> image

Great, go ahead and ask for another review

File: index.html [hunk 1]
```diff
@@ -342,5 +230,37 @@ <h4 class="ab-titles skills"> Skills</h4>
...
+    <a href="${project.sourceCodeLink}" type="button" class="modal-btn-source button-project-card flex-row">
+    <span>See Source</span>
+    <svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
+    <path d="M20.9989 9.46114C20.9989 8.32124 20.6329 7.25043 19.9342 6.31779C20.2004 5.41969 20.2669 4.27979 20.2004 2.96718C20.1671 2.41451 19.7345 2 19.2022 2C18.9028 2 16.3741 2.03454 14.9101 3.38169C13.6458 3.1399 12.315 3.1399 11.0174 3.38169C9.58667 2.03454 7.05802 2 6.7253 2C6.19296 2 5.76042 2.41451 5.72715 2.96718C5.62734 4.27979 5.72715 5.41969 5.99333 6.31779C5.29462 7.28497 4.92863 8.35579 4.92863 9.46114C4.92863 11.8791 6.7253 14.0553 9.45358 15.0915C9.35377 15.2988 9.28723 15.5406 9.22068 15.7824C6.32604 15.4715 4.9619 12.7427 4.89536 12.639C4.66246 12.1209 4.06357 11.9136 3.56449 12.19C3.06541 12.4318 2.86578 13.0535 3.13196 13.5717C3.1985 13.7444 5.02845 17.4404 9.05432 17.8549V20.9637C9.05432 21.5509 9.48686 22 10.0525 22H15.875C16.4406 22 16.8732 21.5509 16.8732 20.9637V16.8532C16.8732 16.2314 16.7401 15.6442 16.5072 15.1261C19.2022 14.0553 20.9989 11.9136 20.9989 9.46114Z" fill="white"/>
+    </svg></a>
+  </div>
+</div> -->
```
Comment (abdulhamiid) [inline index.html:266]: Please, let's not commit old pieces of code as inline comments. They will make our project look messy. If you need to review a previous version of your code, you can always use git history. 👍🤗

File: js/popup.js [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+        </div>
+        <div class="list-technologies-card">
+          <ul class="list-technologies-card-ul flex-row">
+            ${technologiesList}
+          </ul>
+        </div>
+        <button open-modal type="button" class="button-project-card flex-row">See project</button>
+      `;
+    worksSection.appendChild(card);
+  });
```
Comment (abdulhamiid) [inline js/popup.js:105]: Your design (Good job implementing the hover state for the project cards. 😊👍)
image

Figma
image

However, Looking at the design on [Figma](https://www.figma.com/file/l7SqJ3ZfkAKih9sFxvWSR4/Microverse-Student-Project-1?node-id=0%3A1), it is when the user hovers over each card that the button `See Project` appears. please let's correct that. 🤗🛠

**Review by abdulhamiid — CHANGES_REQUESTED** [review]
# Status: Required Changes :recycle:
Hi @Rebelzob 👋, 
Alt Text
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights 🌟

- ✔️ Descriptions are descriptive
- ✔️ Create UIs adaptable to different screen sizes using media queries.
- ✔️ Parse a Figma design to create a UI.

## Required Changes ♻️
_Please kindly check the comments under the review._

## Optional suggestions 🤷‍♂️
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: js/popup.js [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    <ul class="list-technologies-ul modal-techs-btn flex-row">
+      ${technologiesList}
+    </ul>
+  </div>
+  </div>
+  <div class="modal-desktop-bottom">
+  <img class="modal-img item1" src="${project.modalImagePath}" alt="snapshot">
+  <div class="modal-paragraph item2">
+    <p>${project.modalDescription}</p>
+  </div>
```
[the remark this answers is not in the document]
Comment (Rebelzob) [reply js/popup.js:136]: Hello @abdulhamiid and thanks for the review, by the way, I can see modal desktop version is different from yours in my browser 

image

I still need to add the blur which I will, but don't know what to do about the desktop being so different yours from mine, what do you suggest?

[the remark this answers is not in the document]
Comment (Rebelzob) [reply js/popup.js:136]: also, the dialog tag is widely supported by most the browsers 

https://caniuse.com/?search=dialog

**Review by Reem-lab — APPROVED** [review]
# Hi Team,  👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights
- [ ] Great job on the funtionality for the pop up💯
- [ ] Followed GitHub flow 🔥
- [ ] No Linters error ✅
- [ ] Good job working on the design 🔥⭐

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**